lib/mongo/operation/shared/write.rb

```diff
@@ -37,13 +37,7 @@ module Write
       def execute(server, context:)
         server.with_connection(service_id: context.service_id) do |connection|
           validate!(connection)
-          op = if connection.features.op_msg_enabled?
-              self.class::OpMsg.new(spec)
-            elsif !acknowledged_write?
-              self.class::Legacy.new(spec)
-            else
-              self.class::Command.new(spec)
-            end
+          op = self.class::OpMsg.new(spec)
```

Comment:
This is correct but for a different reason than I stated in our conversation - looking at this code again, I'm assuming the reason why Legacy was used was because the Command protocol doesn't have a way of specifying how to perform an unacknowledged write (i.e. it doesn't have a provision for the respective flag).

Reply:
Ah gotcha!